TestProjects/ShaderGraph/Assets/CommonAssets/Editor/ShaderGraphTestRenderer.cs

```diff
@@ -0,0 +1,261 @@
+            wrongImageCount++;
+            int wrongPixels = Math.Abs(incorrectPixels - expectedIncorrectPixels);
+            if (wrongPixels > mostWrongPixels)
+            {
+                mostWrongPixels = wrongPixels;
+                mostWrongString = $"{filePrefix} incorrect pixels expected: {expectedIncorrectPixels} actual: {incorrectPixels}";
+            }
+        }
+
+        RenderTexture.ReleaseTemporary(target);
+        return incorrectPixels;
+    }
+
+    internal static Shader BuildShaderGraph(GraphData graph, string name, bool hide = true)
+    {
+        var generator = new Generator(graph, graph.outputNode, GenerationMode.ForReals, "TransformGraph", null);
+        string shaderString = generator.generatedShader;
+
+        var shader = ShaderUtil.CreateShaderAsset(shaderString, false);
+        if (hide)
+            shader.hideFlags = HideFlags.HideAndDontSave;
+
+        return shader;
+    }
+
+    internal static void RenderMeshWithMaterial(Camera cam, Mesh mesh, Matrix4x4 transform, Material mat, RenderTexture target, bool useSRP = true)
+    {
+        // Force async compile OFF
+        var wasAsyncAllowed = ShaderUtil.allowAsyncCompilation;
+        ShaderUtil.allowAsyncCompilation = false;
+
+        var previousRenderTexture = RenderTexture.active;
+        RenderTexture.active = target;
+
+        GL.Clear(true, true, Color.black);
+
+        cam.targetTexture = target;
+        Graphics.DrawMesh(
+            mesh: mesh,
+            matrix: transform,
+            material: mat,
+            layer: 1,
+            camera: cam,
+            submeshIndex: 0,
+            properties: null,
+            castShadows: ShadowCastingMode.Off,
+            receiveShadows: false,
+            probeAnchor: null,
+            useLightProbes: false);
+
+        var previousUseSRP = Unsupported.useScriptableRenderPipeline;
+        Unsupported.useScriptableRenderPipeline = useSRP;
+        cam.Render();
+        Unsupported.useScriptableRenderPipeline = previousUseSRP;
+
+        RenderTexture.active = previousRenderTexture;
+        ShaderUtil.allowAsyncCompilation = wasAsyncAllowed;
+    }
+
+    internal static void CreateDirectoriesForFilePath(string systemFilePath)
+    {
+        var dirPath = Path.GetDirectoryName(systemFilePath);
+        CreateDirectories(dirPath);
+    }
+
+    internal static void CreateDirectories(string systemDirectoryPath)
+    {
+        //var systemDirectory = Application.dataPath + "/../" + directoryPath;
+        Directory.CreateDirectory(systemDirectoryPath);
+        /*
+        var dirs = directoryPath.Split('/');
+        string curpath = string.Empty;
+        foreach (var dir in dirs)
+        {
+            var parentPath = curpath;
+            curpath = curpath + dir;
+            if (!UnityEditor.AssetDatabase.IsValidFolder(curpath))
+            {
+                AssetDatabase.CreateFolder(parentPath, dir);
+            }
+            curpath = curpath + '/';
+        }
```

Comment:
Is the commented out code just for reference?  Or something we might need for anticipated test framework chnages?

Reply:
Oh, I can delete this.  Turns out there's a single function that can create an entire path all at once at the system level..   Don't really need this function.  :)